### PR TITLE
Make sure BuildRequest has low_priority_node_selector set

### PR DESCRIPTION
**Pull request by vrutkovs**
Related: #556

Since the whole class is flexmocked in tests this has slipped: the class instance 
should store the received low_priority_node_selector

Signed-off-by: Vadim Rutkovsky <vrutkovs@redhat.com>

### CODE REVIEW

File: osbs/build/build_request.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def set_params(self, **kwargs):
             pass
 
         self.base_image = kwargs.get('base_image')
+        self.low_priority_node_selector = kwargs.get('low_priority_node_selector')
```
Comment (twaugh) [inline osbs/build/build_request.py]: How about updating the docstring?

Comment (vrutkovs) [inline osbs/build/build_request.py]: Updated

File: tests/test_api.py [hunk 2]
```diff
@@ -1506,6 +1489,69 @@ def verify_build_json(passed_build_json):
...
+                    [general]
+                    build_json_dir = {inputs}
+                    [default]
+                    low_priority_node_selector = {nodeselector}
+                    """.format(inputs=INPUTS_PATH, nodeselector=nodeselector_str)))
+                fp.flush()
+                config = Configuration(fp.name)
+        else:
+            config = Configuration()
+        osbs = OSBS(config, config)
```
Comment (vrutkovs) [inline tests/test_api.py]: Flake8 doesn't know its being used in `flexmock(osbs.os)` later on, weird

**Comment by vrutkovs** [comment]
Updated the PR - added a new test which checks that `_create_scratch_build` renders a correct Build object with `nodeSelector` set if config has an option for it

File: tests/test_api.py [hunk 3]
```diff
@@ -1506,6 +1489,68 @@ def verify_build_json(passed_build_json):
...
```
Comment (lcarva) [inline tests/test_api.py]: Oh that's a good catch! @vrutkovs can you address this?

File: tests/test_api.py (hunk 2 above)
Comment (lcarva) [inline tests/test_api.py]: Think it's complaining that the `osbs` in line 45 is not used:
```python
45: from tests.fake_api import openshift, osbs, osbs106, osbs_cant_orchestrate
```

Comment (vrutkovs) [inline tests/test_api.py]: Added `noqa` to this line - these are test fixtures and actually are used.

I've fixed other pep8 errors in this file

**Comment by vrutkovs** [comment]
All flake8 errors in this file have been fixed

**Comment by lcarva** [comment]
LGTM, but is coverall stuck?

**Comment by vrutkovs** [comment]
Coveralls has been stuck for a day for master commits also - https://coveralls.io/stories

**Pull request merged**